spec/requests/composer_controller_spec.rb

```diff
@@ -0,0 +1,250 @@
+# frozen_string_literal: true
+
+RSpec.describe ComposerController do
+  describe '#mentions' do
+    fab!(:user) { Fabricate(:user) }
+
+    fab!(:group) { Fabricate(:group, messageable_level: Group::ALIAS_LEVELS[:everyone], mentionable_level: Group::ALIAS_LEVELS[:everyone]) }
+    fab!(:invisible_group) { Fabricate(:group, visibility_level: Group.visibility_levels[:owners]) }
+    fab!(:unmessageable_group) { Fabricate(:group, messageable_level: Group::ALIAS_LEVELS[:nobody], mentionable_level: Group::ALIAS_LEVELS[:everyone]) }
+    fab!(:unmentionable_group) { Fabricate(:group, messageable_level: Group::ALIAS_LEVELS[:everyone], mentionable_level: Group::ALIAS_LEVELS[:nobody]) }
+
+    before do
+      sign_in(Fabricate(:user))
+    end
+
+    context 'without a topic' do
+      it 'finds mentions' do
+        get '/composer/mentions.json', params: {
+          names: [
+            'invaliduserorgroup',
+            user.username,
+            group.name,
+            invisible_group.name,
+            unmessageable_group.name,
+            unmentionable_group.name
+          ],
+        }
+
+        expect(response.status).to eq(200)
+
+        expect(response.parsed_body['users']).to contain_exactly(user.username)
+        expect(response.parsed_body['user_reasons']).to contain_exactly()
```

Comment:
I believe this will return a hash and we should assert for that instead since `contain_exactly` doesn't actually indicate what the object should be.